hack/tools/release/testimage.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function TESTIMAGE() {
+  IMAGE="${1}"
+  ARCH="${2}"
+  GREPGOPATH="${3}"
+  BINARYPATH="${4:-manager}"
+
+  echo "> Testing $IMAGE"
+
+  docker save "${IMAGE}" -o img.tar
+  mkdir -p extracted-img extracted
+  tar -xf img.tar -C extracted-img/
+  while IFS= read -r -d '' layer
+  do
+    tar -xf "${layer}" -C extracted
+  done <   <(find extracted-img/ -name "*.tar" -print0)
+
+  go version -m extracted/"${BINARYPATH}" | grep -P "\tpath" | grep -E -q -e "${GREPGOPATH}" || (echo "FAILED ${IMAGE} expected value for path: \"${GREPGOPATH}\""; go version -m extracted/"${BINARYPATH}" | grep -P "\tpath")
+  go version -m extracted/"${BINARYPATH}" | grep -q -E "GOARCH=${ARCH}$" || (echo "Failed ${IMAGE} expected GOARCH to be \"$ARCH\""; go version -m extracted/"${BINARYPATH}" | grep "GOARCH=")
+
+  rm -rf img.tar extracted-img extracted
```

Comment:
For using the tmp directory:

```suggestion
  docker save "${IMAGE}" -o ${_tmp}/img.tar
  mkdir -p ${_tmp}/extracted-img ${_tmp}/extracted
  tar -xf ${_tmp}/img.tar -C ${_tmp}/extracted-img/
  while IFS= read -r -d '' layer
  do
    tar -xf "${layer}" -C ${_tmp}/extracted
  done <   <(find ${_tmp}/extracted-img/ -name "*.tar" -print0)

  go version -m ${_tmp}/extracted/"${BINARYPATH}" | grep -P "\tpath" | grep -E -q -e "${GREPGOPATH}" || (echo "FAILED ${IMAGE} expected value for path: \"${GREPGOPATH}\""; go version -m ${_tmp}/extracted/"${BINARYPATH}" | grep -P "\tpath")
  go version -m ${_tmp}/extracted/"${BINARYPATH}" | grep -q -E "GOARCH=${ARCH}$" || (echo "Failed ${IMAGE} expected GOARCH to be \"$ARCH\""; go version -m ${_tmp}/extracted/"${BINARYPATH}" | grep "GOARCH=")

  rm -rf ${_tmp}/img.tar ${_tmp}/extracted-img ${_tmp}/extracted
```